hs_core/models.py

```diff
@@ -4747,3 +4749,26 @@ def resource_creation_signal_handler(sender, instance, created, **kwargs):
 def resource_update_signal_handler(sender, instance, created, **kwargs):
     """Do nothing (noop)."""
     pass
+
+
+class SOLRQueue(models.Model):
+    """
+    This implements an update queue for SOLR resources.
+    It contains a set of resources -- one element per resource -- that should be updated in SOLR
+    for changes in Django.
+    """
+    resource = models.ForeignKey(BaseResource, unique=True)
+
+    @classmethod
+    def add(cls, resource):
+        """ add a resource to the update queue without duplicates """
+        with transaction.atomic():
+            cls.objects.get_or_create(resource=resource)
+
+    @classmethod
+    def read_and_clear(cls):
+        """ read a list of resources to update -- order unimportant -- and clear the queue """
+        with transaction.atomic():
+            everything = [x.resource for x in cls.objects.all()]  # force aggressive evaluation
+            cls.objects.all().delete()
```

Comment:
SOLRQueue is cleared even when an error occurs that stops processing the rest of the resources.